### DISCUSSION TITLE
Windows Forms Designer debugging in NetCore

### DISCUSSION THREAD

**Discussion by SenthilKumaranRajan** [dotnet/winforms#4894] [Q&A] [answered]
We could not able to perform design time debugging in NetCore Windows Forms.

We have tried to perform debugging as we perform for dot NetFramework but our source code cannot be able to debug at design time. We have followed the below steps.

1.	Set the breakpoint in our custom control constructor.
2.	Set the start-up project as “devenv.exe”.
 
![image](https://user-images.githubusercontent.com/44424557/117460626-57d67a00-af6a-11eb-9367-e7d823ebee11.png)

3.	On running the application it shows another VisualStudio instance where we will select the project that we need to debug.
4.	Then on opening the designer and adding the custom control from the toolbox.

Expected result:
                The Breakpoint set at our custom control constructor should hit.

Observed result:
                The breakpoint hit properly in the .Net Framework project. But not for the Net core project.

**Please let us know if we need to enable any other operations to debug for the NET core project.**

Note:
We have enabled the preview feature for NetCore.
 
![image](https://user-images.githubusercontent.com/44424557/117460590-4ee5a880-af6a-11eb-89e2-66e43746b787.png)

**Comment by RussKie** [comment, marked as answer]
The issue is that your control designer runs a different process than devenv.exe because it needs to run on .NET Core, while Visual Studio runs on .NET Framework. To debug, do exactly as you did below. However, before adding your control from the designer, attach to the "DesignToolsServer.exe" process from the instance of Visual Studio that is doing the debugging.

**Reply by SenthilKumaranRajan** [reply]
Works fine thanks :)
